Support lts/-n aliases

jablko opened the pull request:
**Description:**
https://github.com/actions/setup-node/pull/270 added support for `lts/<alias>` and `lts/*` nvm aliases. This adds support for nvm's (and hence [`.nvmrc`/`node-version-file`](https://github.com/actions/setup-node/pull/338)'s) `lts/-n`: https://github.com/nvm-sh/nvm/issues/1998#issuecomment-594958684

**Related issue:**
Add link to the related issue.

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

e-korolevskii commented:
Hi, @jablko,

You need to manually run build and add the results to the PR.

Cheers.

jablko commented:
- [x] Thank you @e-korolevskii! I've added the build products to the commit.

jablko commented:
- [x] @dmitry-shibanov Thanks for reviewing! I've now added an e2e test.

jablko commented:
@dmitry-shibanov I noticed:
1. The e2e test exposed a mistake I made and have now corrected.
2. The lts-syntax e2e test was missing a "Verify node and npm" step like the other versions.yml jobs, so I added that too.

jablko commented:
I've rebased and combined the repetitive LTS tests with `it.each(table)` --- but I'm happy to leave them unrolled if you prefer?

dmitry-shibanov commented:
Hello @jablko. Could you please sync with the main branch ?

jablko commented:
- [x] @dmitry-shibanov Yes, I've rebased this.

dmitry-shibanov commented:
Hello @jablko. I think we should not reverse initial manifest. If we specify check-latest to true the action won't return the latest lts version for lts aliases. 
 - https://github.com/actions/setup-node/runs/6659118565?check_suite_focus=true#step:3:10

The [code](https://github.com/actions/setup-node/blob/c4f9da006e738880a70a0c2b9f289e4a9424db2a/src/installer.ts#L253) return the major version that is why when searching for a satisfied version in the reversed array, not the latest version will be taken.

jablko commented:
Thanks @dmitry-shibanov, you're right, `tc.findFromManifest()` [depends on the manifest's order](https://github.com/actions/toolkit/blob/4e9375da095a1d1f59b3314589fab6cd0d215139/packages/tool-cache/src/manifest.ts#L82-L105).

- [x] I've added an entry to the unit tests' mock manifest, so one of its major versions has at least two minor versions, which exposes my mistake. Then I removed the `.reverse()` and confirmed the tests pass again and [we get the max satisfying version](https://github.com/actions/setup-node/runs/6662212311?check_suite_focus=true#step:3:10) when `check-latest` is true.
